### ISSUE TITLE
Stateless Install: deis-router continually logs `signal process started`

### ISSUE THREAD

**Issue by LoicMahieu** [deis/deis#4447]
Stateless deis install: 1.10.0

Each 5 seconds, the router logs `signal process started` even if no changes are made.

```
Sep 11 17:38:34 ip1 logger: 2015-09-11T15:38:34UTC deis-router[1]: [INFO] - 2015/09/11 15:38:34 [notice] 10050#0: signal process started
Sep 11 17:38:34 ip2 logger: 2015-09-11T15:38:34UTC deis-router[1]: [INFO] - 2015/09/11 15:38:34 [notice] 7319#0: signal process started
Sep 11 17:38:34 ip3 logger: 2015-09-11T15:38:34UTC deis-router[1]: [INFO] - 2015/09/11 15:38:34 [notice] 1648#0: signal process started
Sep 11 17:38:39 ip1 logger: 2015-09-11T15:38:39UTC deis-router[1]: [INFO] - 2015/09/11 15:38:39 [notice] 10060#0: signal process started
Sep 11 17:38:39 ip2 logger: 2015-09-11T15:38:39UTC deis-router[1]: [INFO] - 2015/09/11 15:38:39 [notice] 7329#0: signal process started
Sep 11 17:38:39 ip3 logger: 2015-09-11T15:38:39UTC deis-router[1]: [INFO] - 2015/09/11 15:38:39 [notice] 1658#0: signal process started
Sep 11 17:38:44 ip1 logger: 2015-09-11T15:38:44UTC deis-router[1]: [INFO] - 2015/09/11 15:38:44 [notice] 10070#0: signal process started
Sep 11 17:38:44 ip2 logger: 2015-09-11T15:38:44UTC deis-router[1]: [INFO] - 2015/09/11 15:38:44 [notice] 7339#0: signal process started
Sep 11 17:38:44 ip3 logger: 2015-09-11T15:38:44UTC deis-router[1]: [INFO] - 2015/09/11 15:38:44 [notice] 1668#0: signal process started
Sep 11 17:38:49 ip1 logger: 2015-09-11T15:38:49UTC deis-router[1]: [INFO] - 2015/09/11 15:38:49 [notice] 10080#0: signal process started
Sep 11 17:38:49 ip2 logger: 2015-09-11T15:38:49UTC deis-router[1]: [INFO] - 2015/09/11 15:38:49 [notice] 7349#0: signal process started
Sep 11 17:38:49 ip3 logger: 2015-09-11T15:38:49UTC deis-router[1]: [INFO] - 2015/09/11 15:38:49 [notice] 1678#0: signal process started
```

I don't know if it is problematic but I found an old issue #2030 about that.

**Comment by carmstrong** [comment]
This is confd reloading nginx anytime it detects that a configuration file has changed. This makes me think that one of your components (like controller) is flapping, and router is re-templating it each time. Do you see anything to correlate in any other logs?

**Comment by LoicMahieu** [comment]
I can't see anything in other logs... I quickly setup a Deis in stateless on DO and I can't replicate.
I will re-install my cluster in the next day. Due to #4419 some stateful components have been installed and even if I removed all components (fleet units and docker containers), I still see some ceph related logs: 
```
...
Sep 13 15:37:42 deis-1 kernel: libceph: mon2 10.10.8.12:6789 socket closed (con state CONNECTING)
...
```

In waiting a clean re-install, I close the issue.

**Comment by Crispy1975** [comment]
@carmstrong we are experiencing this at the moment.

**Comment by bacongobbler** [comment]
@Crispy1975 is one of your components/apps flapping? That may cause such an issue like this to occur.

**Comment by Crispy1975** [comment]
@bacongobbler Not that I can see here, bar getting terminated connections which I think is due to the router all is normal.

**Comment by Crispy1975** [comment]
@bacongobbler ok, restart the controller, builder, and did a rolling restart on the routers, but still have one host node that is currently out out of service with our ELB due to nginx restarting constantly.

This is actually happening on all three cluster nodes however one is significantly worse than the others. Thankfully the two that are live are passing ELB health checks.

Any ideas on this as it's something of a major issue.

**Comment by LoicMahieu** [comment]
These logs are back after a graceful upgrade from stateless 1.10.0 to 1.11.1
```
Oct 15 18:14:04 x.x.x.x logger: 2015-10-15T16:14:04UTC deis-router[1]: [INFO] - 2015/10/15 16:14:04 [notice] 1639#0: signal process started
Oct 15 18:14:09 x.x.x.x logger: 2015-10-15T16:14:09UTC deis-router[1]: [INFO] - 2015/10/15 16:14:09 [notice] 1649#0: signal process started
Oct 15 18:14:34 x.x.x.x logger: 2015-10-15T16:14:34UTC deis-router[1]: [INFO] - 2015/10/15 16:14:34 [notice] 1659#0: signal process started
Oct 15 18:14:39 x.x.x.x logger: 2015-10-15T16:14:39UTC deis-router[1]: [INFO] - 2015/10/15 16:14:39 [notice] 1669#0: signal process started
Oct 15 18:15:05 x.x.x.x logger: 2015-10-15T16:15:04UTC deis-router[1]: [INFO] - 2015/10/15 16:15:04 [notice] 1685#0: signal process started
Oct 15 18:15:10 x.x.x.x logger: 2015-10-15T16:15:10UTC deis-router[1]: [INFO] - 2015/10/15 16:15:10 [notice] 1695#0: signal process started
Oct 15 18:15:15 x.x.x.x logger: 2015-10-15T16:15:15UTC deis-router[1]: [INFO] - 2015/10/15 16:15:15 [notice] 1068#0: signal process started
Oct 15 18:15:21 x.x.x.x logger: 2015-10-15T16:15:20UTC deis-router[1]: [INFO] - 2015/10/15 16:15:20 [notice] 1078#0: signal process started
Oct 15 18:15:25 x.x.x.x logger: 2015-10-15T16:15:25UTC deis-router[1]: [INFO] - 2015/10/15 16:15:25 [notice] 1705#0: signal process started
Oct 15 18:15:26 x.x.x.x logger: 2015-10-15T16:15:25UTC deis-router[1]: [INFO] - 2015/10/15 16:15:25 [notice] 1088#0: signal process started
...
```
I tried to restart routers, controller and builder but it does not solves the problem.
Tell me about I can give you more information. Thanks.

**Comment by carmstrong** [comment]
@LoicMahieu From the other component logs, does it look like any of them are restarting? This means that something in the nginx config is changing.

**Comment by LoicMahieu** [comment]
I had a strange issue... `deis apps` was randomly warning:
```
!    WARNING: Client and server API versions do not match. Please consider upgrading.
!    Client version: 1.7
!    Server version: 1.6
```

**Comment by LoicMahieu** [comment]
Oh.... `controller` and `builder` containers of `1.10.0` are still running !!! But are not visible on fleet ! Arg...
I will stop them, I think it is the problem... Each components are re-setting their keys.
For the version problem, I suppose the routers to switch on the old and new controller...

**Comment by carmstrong** [comment]
> I will stop them, I think it is the problem... Each components are re-setting their keys.
> For the version problem, I suppose the routers to switch on the old and new controller..

That is exactly what's happening. Once you stop the old versions, everything should work fine. This was fixed in #4414 so you shouldn't run into it on future upgrades. Sorry for the trouble!

**Comment by LoicMahieu** [comment]
Old controller and builder are deleted and logs are gone...
Thanks @carmstrong for response and thanks team for already have been fixed that :D Awesome!

**Comment by kkola** [comment]
I recently upgraded to the 1.13.0 LTS release from 1.12.3 and everything was fine until I `deisctl config router set bodySize=1000m` and restarted my routers.
The apps that I deploy to Deis seem to be running fine and all other components seem to be reporting the usual log messages, but I keep getting these messages in the logs from all routers.

`Apr 13 01:06:15 ip-10-21-1-219.ec2.internal sh[17188]: [INFO] - 2016/04/13 01:06:15 [notice] 403#0: signal process started
...
...
...`

Is this something that we need to be concerned about? What other log locations should I be looking into?

**Comment by krancour** [comment]
Those messages are Nginx restarting in response to changed configuration.  The most likely explanation is that etcd has list quorum or has become split-brained.  Consider looking into the health of the underlying etcd cluster.

**Comment by kkola** [comment]
My cluster reports that it's healthy. When I run the command `etcdctl cluster-health` this is the output I get:

> core@ip-10-21-1-219 ~ $ etcdctl cluster-health
member 3816712910dec058 is healthy: got healthy result from http://10.21.2.33:2379
member 67134f7aef2a3c06 is healthy: got healthy result from http://10.21.2.32:2379
member bc4e0de7c9e002b6 is healthy: got healthy result from http://10.21.1.219:2379
cluster is healthy

**Comment by kkola** [comment]
I was able to figure out what went wrong. Apparently one of the Docker containers that we deployed was exposing two HTTP ports and this was leading to etcd keys being updated every second or so about a new port mapping for the containers. This basically meant that confd was updating nginx.conf every few seconds if not more frequently. Once we destroyed these apps and changed the associated Dockerfile to only use one port, the error/notice messages just went away from all routers! I just wanted to capture this here for anyone who stumbles across this issue.
